NitrokeyManager.cc

```diff
@@ -390,6 +390,69 @@ using nitrokey::misc::strcpyT;
         }
     }
 
+    uint8_t NitrokeyManager::get_pws_slot_count() {
+        if (device == nullptr) {
+            return 0;
+        } else {
+            return PWS_SLOT_COUNT;
+        }
+    }
+
+    size_t NitrokeyManager::get_pws_name_length() {
+        if (device == nullptr) {
+            return 0;
+        } else {
+            return PWS_SLOTNAME_LENGTH;
+        }
+    }
+
+    size_t NitrokeyManager::get_pws_login_length() {
+        if (device == nullptr) {
+            return 0;
+        } else {
+            return PWS_LOGINNAME_LENGTH;
+        }
+    }
+
+    size_t NitrokeyManager::get_pws_password_length() {
+        if (device == nullptr) {
+            return 0;
+        } else {
+            return PWS_PASSWORD_LENGTH;
+        }
+    }
+
+    uint8_t NitrokeyManager::get_totp_slot_count() {
+        if (device == nullptr) {
+            return 0;
+        } else {
+            return 15;
+        }
+    }
+
+    uint8_t NitrokeyManager::get_hotp_slot_count() {
+        if (device == nullptr) {
+            return 0;
+        } else {
+            return 3;
+        }
+    }
+
+    size_t NitrokeyManager::get_otp_name_length() {
+        if (device == nullptr) {
+            return 0;
+        } else {
+            return 15;
+        }
+    }
+
+    size_t NitrokeyManager::get_otp_secret_length() {
+        if (device == nullptr) {
+            return 0;
+        } else {
+            return 40;
```

Comment:
For earlier firmware versions of Nitrokey Pro this would be 20. Please add FIXME here to remember about extending that (details are in the libnitrokey's OTP implementation).

Reply:
Thanks, I’ll have a look at it.

Reply:
Apparently, it was changed in v0.8 for the Nitrokey Pro and v0.54 for the Nitrokey Storage, see `is_320_OTP_secret_supported`.  But it would be nicer if we would not have to perform a `GET_STATUS` command to query this number.  I’ll think about other solutions.